src/postamble.js

```diff
@@ -485,6 +485,14 @@ function exit(status, implicit) {
   quit_(status, new ExitStatus(status));
 }
 
+#if '___wasm_call_ctors' in IMPLEMENTED_FUNCTIONS
+#if !DECLARE_ASM_MODULE_EXPORTS
```

Comment:
why is this needed? even with this option we write `__wasm_call_ctors` into the global scope, so always calling it as on line 492 should be ok?

Reply:
Oh, is the issue that we have not written it out yet?

Could we move this to this code happens after we write it?

Reply:
With NO_DECLARE_ASM_MODULE_EXPORT we don't write `__wasm_call_ctors` (or any wasm exports) to the global scope statically, they get dynamically injected once the module is loaded.  

If we could be sure this code ran after module instantiation we could avoid this.. perhaps we can find a place where that is true, but here in the global scope the module has not yet been instantiated.

Reply:
Am I misunderstanding what is going on?    This is certainly needed to make `NO_DECLARE_ASM_MODULE_EXPORT` tests pass as it stands.

Reply:
I think you are right, now that I follow this.

My concern now is code clarity. At least a comment could help. But I think we can just move this into `receiveInstance` after
```js
#if !DECLARE_ASM_MODULE_EXPORTS
    // If we didn't declare the asm exports as top level enties this function
    // is in charge of programatically exporting them on the global object.
    exportAsmFunctions(exports);
#endif
```
The exports definitely exist after that code.

The only risk is ordering. Maybe other OnInits would have been added meanwhile?

Reply:
I think at addOnIinit does the right thing and puts the thing being added first in the list using `unshift` (I assume we always want it ctors to run first?)

Reply:
I tried that and it didn't work for some reason.. I'll try again to be sure.